src/FrameReflower/Text.php

```diff
@@ -386,13 +389,35 @@ function reflow(BlockFrameDecorator $block = null)
 
         // Determine the text height
         $style = $frame->get_style();
+        $font_metrics = $this->getFontMetrics();
+
+        // Handle text transform and white space
+        $text = $this->pre_process_text($frame->get_text());
+
+        // map text to fonts based on supported Unicode range
+        $families = array_map(
+            function ($value) {
+                return trim($value, " '\"");
+            },
+            explode(",", $style->get_specified("font_family"))
+        );
+        $charMapping = $font_metrics->mapFontsToText($text, $families, $font_metrics->getType($style->font_weight . ' ' . $style->font_style), 1);
+        if (isset($charMapping[0])) {
+            if ($charMapping[0]["length"] !== 0) {
+                $frame->split_text($charMapping[0]["length"]);
+                $text = $this->pre_process_text($frame->get_text());
+            }
+            $mapped_font = $charMapping[0]["font"];
+            if ($mapped_font !== null) {
```

Comment:
Could fall back to the first available font here directly, so it does not need to be determined again below (in line 417 via the `Style` getter). Maybe `mapFontsToText` should directly fall back to the first font instead of `null`?

Reply:
First available specified may not be a valid font-family. We could default to the first valid in the mapping logic. Though then that method does not indicate when text is completely unsupported by the specified fonts. Which I guess is a question of generalization vs. our particular use case.

As you can tell I lean towards generalization where I can. I think this shouldn't be an issue in most cases where:
1. the user is specifying valid fonts
2. the text is supported by one of the specified fonts

And even if we do have to fall back to the Style class logic, so long as there aren't a lot of splits that require falling back the performance impact should be minimized.

Reply:
Yeah, it's quite the edge case. I'll think on this more while I think on some of the other things you brought up.

Reply:
I'm going to table further improvements for now.